### PR TITLE
Update to 1.18.2

### PR DESCRIPTION
**Pull request by IMS212**
Ready.

### CODE REVIEW

File: src/main/java/net/coderbot/iris/mixin/MixinLevelRenderer.java [hunk 1]
```diff
@@ -123,33 +124,33 @@
...
 	@Inject(method = "renderSky", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/player/LocalPlayer;getEyePosition(F)Lnet/minecraft/world/phys/Vec3;"))
-	private void iris$setVoidRenderStage(PoseStack poseStack, Matrix4f matrix4f, float f, Runnable runnable, CallbackInfo ci) {
+	private void iris$setVoidRenderStage(PoseStack poseStack, Matrix4f matrix4f, float f, Camera camera, boolean bl, Runnable runnable, CallbackInfo ci) {
 		pipeline.setPhase(WorldRenderingPhase.VOID);
 	}
 
 	@Inject(method = "renderSky", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/multiplayer/ClientLevel;getTimeOfDay(F)F"),
 			slice = @Slice(from = @At(value = "FIELD", target = "com/mojang/math/Vector3f.YP : Lcom/mojang/math/Vector3f;")))
-	private void iris$renderSky$tiltSun(PoseStack poseStack, Matrix4f projectionMatrix, float f, Runnable runnable, CallbackInfo callback) {
+	private void iris$renderSky$tiltSun(PoseStack poseStack, Matrix4f matrix4f, float f, Camera camera, boolean bl, Runnable runnable, CallbackInfo ci) {
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/mixin/MixinLevelRenderer.java:153]: matrix4f -> projectionMatrix?

File: src/main/java/net/coderbot/iris/uniforms/FogUniforms.java [hunk 2]
```diff
@@ -18,6 +19,7 @@ public static void addFogUniforms(DynamicUniformHolder uniforms, FogMode fogMode
 		if (fogMode == FogMode.OFF) {
 			uniforms.uniform1f(UniformUpdateFrequency.ONCE, "fogDensity", () -> 0.0F);
 			uniforms.uniform1i(UniformUpdateFrequency.ONCE, "fogMode", () -> 0);
+			uniforms.uniform1i(UniformUpdateFrequency.ONCE, "fogShape", FogShape.CYLINDER::getIndex);
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/uniforms/FogUniforms.java:22]: I don't like the idea of not defining what this means ourselves & just passing through the vanilla enum, especially if the vanilla enum changes.

File: src/main/java/net/coderbot/iris/uniforms/FogUniforms.java [hunk 3]
```diff
@@ -33,6 +35,8 @@ public static void addFogUniforms(DynamicUniformHolder uniforms, FogMode fogMode
 					return GL11.GL_EXP2;
 				}
 			}, notifier -> {});
+
+			uniforms.uniform1i(PER_FRAME, "fogShape", RenderSystem.getShaderFogShape()::getIndex);
```
Comment (coderbot16) [inline src/main/java/net/coderbot/iris/uniforms/FogUniforms.java:39]: Ditto.

**Review by coderbot16 — CHANGES_REQUESTED** [review]

**Comment by IMS212** [comment]
I've fixed the projectionMatrix name everywhere, as well as fogShape by defining 0 as cylindrical and 1 as spherical.

**Comment by coderbot16** [comment]
I think it would make the most sense to define 0 to be spherical and 1 to be cylindrical. The uniform will be set to the default value 0 on older versions of Iris and OptiFine, which matches existing shader behavior - spherical fog.

**Comment by IMS212** [comment]
Sounds good

**Comment by coderbot16** [comment]
I imagine we can leave the default fogShape as 0 when fog is disabled, not that shaders should be relying on any value when fog is disabled.

**Comment by IMS212** [comment]
Some shaders may depend on fogShape in the future when there’s no fog, so not sure. It might be safe to just keep it like this

**Comment by coderbot16** [comment]
If there's no fog why would they need the shape of it?

**Comment by IMS212** [comment]
Shader’s like Xordev’s default may end up assuming fogShape == 0 means “in water” (bad assumption, but still possible), so fogShape == 1 would counter that.

**Comment by coderbot16** [comment]
Well fogShape means whatever we want it to mean so we don't have to deal with that kind of legacy weirdness

**Comment by IMS212** [comment]
Can we default to -1 for no fog?

**Comment by coderbot16** [comment]
That could work

**Comment by IMS212** [comment]
This is scheduled to be merged on Feb. 27, one day before the release of 1.18.2.

**Comment by coderbot16** [comment]
This doesn't look mergeable - it conflicts with the development branch & we don't know when the next Sodium version will be out.

**Comment by IMS212** [comment]
Resolved the conflict, but Sodium is still an issue, yes.

**Review by coderbot16 — COMMENTED** [review]
The changes here look fine but this cannot be merged until Sodium is updated. I guess people will just have to wait a bit to use shaders on 1.18.2, not a huge deal overall.

**Comment by IMS212** [comment]
Should we still have this merged, or just keep the 1.18.2 branch as-is?

**Comment by coderbot16** [comment]
We can keep the 1.18.2 branch and remove the 1.18 branch, since it's been superseded.

**Pull request closed without merging**